src/Routing/Route/Route.php

```diff
@@ -669,7 +669,7 @@ public function match(array $url, array $context = []): ?string
             $hostOptions['_base'] = $context['_base'];
         }
 
-        $query = !empty($url['?']) ? (array)$url['?'] : [];
+        $query = empty($url['?']) ? [] : (array)$url['?'];
```

Comment:
This is the kind of thing that looks weird:
Why would the primary thing be at the end instead of the first?

    checkExistence ? primary : fallback
    
IMO makes it more logical from normal flow perspective.

Reply:
I think its the problem with the empty, because empty is negative check so to do what you described you need to negate the negative check. 

This may add to cognitive load when you are reading the code when you are not used to it.

Reply:
I am still on the fence here calling !empty a negative one.
It always is read as `if (x) then x otherwise y` IMO.
It now reads as `if (!x) then y otherwise x` which is just weird.

Reply:
In all other cases this would be an eazy solution, but "empty" is complicating this :D

Maybe it is because rector dislikes "empty". PHPStan with strict rules is not allowing "empty" either. (https://www.beberlei.de/post/when_to_use_empty_in_php_i_say_never)

So maybe because of this many of these rules are not working 100%. But on the other hand, replacing empty with strict checks would destroy readability of the code.

Reply:
I am for ignoring empty for the reasons above.
@ADmad ?

Reply:
We can change this to `$query = $url['?'] ?? [];`.  The array casting is not necessary. It needs to be an array. Turning a string `'foo'` to `['foo']` isn't helpful.

Reply:
I think he has problem with all the affected ternary operations that use empty

Reply:
Only a couple of ternary using `empty()` seems to remain after making the changes I suggested. We can address them separately by rewriting the code or setting defaults for the keys.

Reply:
empty() is fine.
I was referring to the yoda ordering for !empty() which in normal speaking/thinking is not negative.